### DISCUSSION TITLE
GitHub Action self hosted runner selection

### DISCUSSION THREAD

**Discussion by CKT021** [community/community#14546] [Actions]
How to use same self-hosted runner across multiple jobs. In my case i have 2 self hosted runner with same labels and if i run workflow some jobs are runnering on 1 and some are running on 2nd self-hosted runner

**Comment by Akash1134** [comment]
Hi @CKT021 ,

Welcome to the community! 

Multiple Self-Hosted Runners with the same labels will pick up the queued jobs based on their availability. 

Are you running all jobs on one runner for a specific reason? We recommend providing an array of labels that includes additional labels as needed to run a job specifying the a same runner. When you specify a label, jobs will be queued on runner that has the label you specified.

Example: Using labels for runner selection
`runs-on: [self-hosted, linux]
`

For more information, see [Workflow syntax for GitHub Actions](https://docs.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idruns-on).

**Reply by CKT021** [reply]
@Akash1134 My use case is I have a single workflow with multiple jobs and multiple instance of runners in a single VM. If i trigger 2 flows my jobs are running with mismatch runners like job1(WF1)-->runner1 job2(WF1)-->runner2.
